uwblueprint/paramedics-react
# Add Location Pin

[Notion ticket](https://www.notion.so/uwblueprintexecs/396370c5e09f4c28bc7f455a7e84c908?v=3d4d369c668e4d2581a2517666768b44&p=03d6681d60694dbeb4612d088595b3f9)

[Figma link](https://www.figma.com/file/IpOPyv79gBwEohvOJvPV6f/%F0%9F%9A%91-Designs-for-Development-S20-F20?node-id=1174%3A21019)

## Changes
- Add pin flow added. This includes:
    - Add pin button
    - Sidebar with autocompletion of addresses (restricted to Canada)
- Enhanced geolocation
- Adding a few more queries
- Minor changes to the hamburger menu due to styling issues

## Screenshots
<img width="1412" alt="Screen Shot 2020-11-13 at 1 33 38 PM" src="https://user-images.githubusercontent.com/21224282/99119004-c509e380-25b5-11eb-8f0b-7c6fc2330539.png">

## Testing
- Test the happy path: make sure that pins are added and open the resulting info window to make sure the information is correct
- Try partially filling in fields and hit cancel. Reopen and all fields should be cleared
- Partially fill fields and tap outside (not hitting cancel). Reopen and fields (except address) should be there
- Try to enter an address outside of Canada. This should not work
- Testing on iPad via tunnelling will not work on this PR because tunnelling wipes the API key
- Check the cache and confirm that it is updating
- Make sure this all works with the hybrid setting 


## Checklist
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

before opening PR

filling out PR

after opening PR
